src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.AvailableItems.targets

```diff
@@ -66,6 +67,9 @@ This item group populates the Build Action drop-down in IDEs.
       <Bind>true</Bind>
       <Pack Condition=" '$(UsingAndroidNETSdk)' == 'true' ">true</Pack>
     </AndroidLibrary>
+    <AndroidJavaSource>
+      <Bind Condition=" '$(_AndroidIsBindingProject)' == 'true' ">true</Bind>
+    </AndroidJavaSource>
```

Comment:
Should we add a new default include for .NET 6 projects here:

https://github.com/xamarin/xamarin-android/blob/02ab8f723b6b8e327a3d8a6fd99192598ca94668/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/Sdk/AutoImport.props#L44

Something like:

```xml
<AndroidJavaSource Include="**/*.java" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
```

And then would we want `%(Bind)` to be `true` or `false` by default?

Reply:
@jonathanpeppers asked:

> And then would we want `%(Bind)` to be `true` or `false` by default?

I think [it depends](https://github.com/xamarin/xamarin-android/pull/5926/files#r638273242):

> The `%(Bind)` attribute will default to `true` in Legacy Java Binding Projects,
> and in .NET 6 projects in which the `$(OutputType)` MSBuild property is `Library`.
> [TODO: verify]
> In Legacy Xamarin.Android Library and Application projects, `%(Bind)` will default to False,
> as well as in .NET 6 projects in which the `$(OutputType)` MSBuild property is `Exe`.

…though now that I read that, 20 minutes later (!), the .NET 6 story is *very confused*: do we really want the default behavior of `@(AndroidJavaSource)` to depend on `$(OutputType)`?  Is this "explainable" or reasonable or sane?

This makes me question the sanity of using `@(AndroidJavaSource)` for both pre-build and post-build actions, with `%(Bind)` as the distinguishing characteristic.

Perhaps we should instead "just" add a new build action `@(AndroidBindJavaSource)`?

*Alternatively*, make `%(AndroidJavaSource.Bind)` a .NET 6+ *only* feature, *not* present in Legacy.  Then, the default could be True by default, which matches `@(AndroidLibrary)`.

This would result in a "semantic break" from Legacy, so this might not be ideal, but it's something to *consider*.  (Maybe?  Or maybe we should just go with a new item group.)

Reply:
@jonathanpeppers [asked](https://github.com/xamarin/xamarin-android/pull/5926/files#r638077508):

> Should we add a new default include for .NET 6 projects here:
> …
> Something like:
> ```xml
> <AndroidJavaSource Include="**/*.java" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
> ```

We need some way of distinguishing between the "bind these Java sources" vs. "include these in the final app" set.  In .NET 6, perhaps this could be done via convention (and maybe also item group, but even a new item group needs to figure out the convention).

Perhaps we should go with a `bind\**\*.java` convention for "Java sources to bind", and either "everything else" *or* `app\**\*.java` for "include these in the final app".

The "everything else" idea would (presumably) simplify `.csproj` migration from Legacy.